src/relations/backend_db_admin.py

```diff
@@ -0,0 +1,198 @@
+
+        if postgres_data.get("master"):
+            dbs["pg_master"] = pgb.parse_kv_string_to_dict(postgres_data.get("master"))
+        else:
+            logger.info("waiting for postgres charm to correctly populate relation data")
+            change_event.defer()
+            return
+
+        standbys_str = postgres_data.get("standbys")
+        standbys = standbys_str.split("\n") if standbys_str else []
+
+        self._update_standbys(cfg, standbys)
+
+        self.charm._render_pgb_config(cfg, reload_pgbouncer=True)
+
+    def _on_relation_departed(self, departed_event: RelationDepartedEvent):
+        """Handle backend-db-admin-relation-departed event.
+
+        Removes unit information from pgbouncer config when a unit is removed.
+        """
+        logger.info("backend database removed - updating config")
+        logger.warning(
+            "DEPRECATION WARNING - backend-db-admin is a legacy relation, and will be deprecated in a future release. "
+        )
+
+        cfg = self.charm._read_pgb_config()
+
+        # Iterate through relation data to get the postgresql unit. Relation data keys are one Unit
+        # and one Application for each side of the relation.
+        event_data = {}
+        for relation_entity, entity_data in departed_event.relation.data.items():
+            if isinstance(relation_entity, Unit) and relation_entity is not self.charm.unit:
+                event_data = entity_data
+                break
+
+        standbys = event_data.get("standbys")
+        standbys = standbys.split("\n") if standbys else []
+
+        self._update_standbys(cfg, standbys)
+
+        self.charm._render_pgb_config(cfg, reload_pgbouncer=True)
+
+    def _update_standbys(self, cfg: PgbConfig, standbys: List[str]):
+        """Updates standby list to match new relation data."""
+        dbs = cfg["databases"]
+
+        standby_names = []
+        for idx, standby in enumerate(standbys):
+            standby_name = f"{STANDBY_PREFIX}{idx}"
+            standby_names.append(standby_name)
+            dbs[standby_name] = pgb.parse_kv_string_to_dict(standby)
+
+        # Remove old standby information
+        for db in list(dbs.keys()):
+            if db[:21] == STANDBY_PREFIX and db not in standby_names:
+                del dbs[db]
+
+        return cfg
+
+    def _on_relation_broken(self, broken_event: RelationBrokenEvent):
+        """Handle backend-db-admin-relation-broken event.
+
+        Removes all traces of this relation from pgbouncer config.
+        """
+        logger.info("backend database removed - updating config")
+        logger.warning(
+            "DEPRECATION WARNING - backend-db-admin is a legacy relation, and will be deprecated in a future release. "
+        )
+
+        cfg = self.charm._read_pgb_config()
+        dbs = cfg["databases"]
+
+        dbs.pop("pg_master", None)
+
+        for db in list(dbs.keys()):
+            # Remove all standbys
+            if db[:21] == STANDBY_PREFIX:
```

Comment:
I didn't get why until the `21`

Reply:
`STANDBY_PREFIX` is 21 characters long, so we're checking if `db` is a backend standby. I've replaced 21 with `len(STANDBY_PREFIX`.